app/controllers/open_access_publications_controller.rb

```diff
@@ -36,16 +36,28 @@ def update
   end
 
   def scholarsphere_file_version
-    extra_params = { journal: publication&.journal&.title }
-    exif_uploads = ScholarsphereExifUploads.new(deposit_params.merge(extra_params))
-
-    if exif_uploads.valid?
-      @cache_files = exif_uploads.cache_files
-      @file_version = exif_uploads.version
-      @file_version_display = exif_uploads.version_display
-      render :scholarsphere_file_version
+    file_version = nil
+    extra_params = { journal: publication&.journal&.title, publication: publication }
+    file_version_uploads = ScholarsphereFileVersionUploads.new(deposit_params.merge(extra_params))
+    @job_ids ||= []
+
+    if file_version_uploads.valid?
+      @cache_files = file_version_uploads.cache_files
+      exif_versions = file_version_uploads.exif_file_versions.compact
+      file_version = ScholarsphereFileVersionUploads.version(exif_versions) if exif_versions.present?
+
+      if file_version == I18n.t('file_versions.published_version')
+        render partial: 'open_access_publications/file_version_result', locals: { file_version: file_version, cache_files: @cache_files }
+      else
+        @cache_files.each do |cache_file|
+          file_version_job = ScholarspherePdfFileVersionJob.perform_later(file_meta: cache_file.to_json, publication_meta: version_check_pub_meta, exif_file_version: file_version)
+          @job_ids.push(file_version_job.job_id)
```

Comment:
I'm also curious about the need to pass around an array of job ID's between HTTP requests in order to check on the status of the jobs as opposed to polling to check on the state resulting from the jobs having been run. In other words, how come we're watching background jobs rather than, for example, checking the state of file upload records in the database?

I'm not saying that this wasn't necessary - I'm really just asking because I don't have quite enough context to understand the reasoning for doing it this way.

Reply:
@banukutlu correct me if I'm wrong, but this job doesn't result in anything changing in the database.  It's just used to prefill the form on the next page.  

The Scholarsphere upload process is now a three-step workflow.  The final step creates the upload record.  The version checking is between the first and second step.

Reply:
> @banukutlu correct me if I'm wrong, but this job doesn't result in anything changing in the database. It's just used to prefill the form on the next page.
> 
> The Scholarsphere upload process is now a three-step workflow. The final step creates the upload record. The version checking is between the first and second step.

Yes, that's correct. It doesn't modify the database or store any results in a field. Instead, it simply populates a form with the file version information, but only if a result is found through the version check. Ultimately, it's up to the user to decide whether to accept the prefilled value or make changes.

Reply:
> I'm also curious about the need to pass around an array of job ID's between HTTP requests in order to check on the status of the jobs as opposed to polling to check on the state resulting from the jobs having been run. In other words, how come we're watching background jobs rather than, for example, checking the state of file upload records in the database (or where ever the jobs' results are being stored)?
> 
Eric, as Alex mentioned, we don't store the status of file version checks in the database. We pass an array of job IDs to keep track of the jobs we need to monitor. We're open to any ideas you may have on this.

Reply:
That makes sense. Thanks for the explanation!